### DISCUSSION TITLE
Best-practices on numeric precision

### DISCUSSION THREAD

**Discussion by jin-zhe** [geoopt/geoopt#191] [Q&A]
Hi just wondering if anyone can provide the best-pracitices (or a rule-of-thumb) on choosing the appropriate numeric precision when working in hyperbolic space?

I have read comments suggesting double-precision for working with neural networks in Poincare space. Is that always necessary?

**Comment by inboxedshoe** [comment]
In my case double precision has been an overall necessity. Without it I notice tensors fall off the manifold quite often and I end up with NaNs. You could try a more aggressive clipping approach but I saw a loss of overall model performance using it. I normally set it through `torch.set_default_dtype(float64)` which makes it easier to test the model with/without it.
